app/controllers/application_controller.rb

```diff
@@ -21,11 +21,11 @@ def authenticate_system(skip_on_duplicated: false)
         update_user_agent
 
         # If SYSTEM_TOKEN_HEADER is present, RMT assumes the client uses a SUSEConnect version
-        # that supports this feature. In this case, refresh the token and include it in the response.
+        # that supports this feature.
         if system_tokens_enabled? && request.headers.key?(SYSTEM_TOKEN_HEADER)
-          @system.update(last_seen_at: Time.zone.now, system_token: SecureRandom.uuid)
+          @system.update(last_seen_at: Time.zone.now)
           headers[SYSTEM_TOKEN_HEADER] = @system.system_token
```

Comment:
could use the new method `system_token_header` here?

Reply:
Shouldn't all of this go to `refresh_system_token`, so that the token only gets rotated `if system_tokens_enabled? && request.headers.key?(SYSTEM_TOKEN_HEADER) == true`?

Reply:
do you mean i should remove the complete if condition from the authenticate_system function from application_controller.rb
change from
   ```
     if system_tokens_enabled? && request.headers.key?(SYSTEM_TOKEN_HEADER)
          @system.update(last_seen_at: Time.zone.now)
          headers[SYSTEM_TOKEN_HEADER] = @system.system_token
          # only update last_seen_at each 3 minutes,
        # so that a system that calls SCC every second doesn't write + lock the database row
        elsif !@system.last_seen_at || @system.last_seen_at < 3.minutes.ago
          @system.touch(:last_seen_at)
        end
```
to
```
        if !@system.last_seen_at || @system.last_seen_at < 3.minutes.ago
          @system.touch(:last_seen_at)
        end
```
add 2)  the below in base_controller.rb so that ( only adding the token in the header )
`  after_action :system_token_header, if: -> { request.headers.key?(SYSTEM_TOKEN_HEADER) }
`
Also, I think, the current way will also work which is already implemented in the PR. without adding after-action in base-controller
as we need to send the token in response headers for all the api's ( read + write ) I think. so for the system namespace controllers, we are always calling authenticate_system function in before-action.

    ```
    if system_tokens_enabled? && request.headers.key?(SYSTEM_TOKEN_HEADER)
          @system.update(last_seen_at: Time.zone.now)
          headers[SYSTEM_TOKEN_HEADER] = @system.system_token
          # only update last_seen_at each 3 minutes,
        # so that a system that calls SCC every second doesn't write + lock the database row
        elsif !@system.last_seen_at || @system.last_seen_at < 3.minutes.ago
          @system.touch(:last_seen_at)
        end
```

Reply:
You're right, we always should return the `SYSTEM_TOKEN_HEADER`, when tokens are enabled.